pkg/util/docker/global.go

```diff
@@ -60,6 +60,9 @@ func HostnameProvider(hostName string) (string, error) {
 
 // ContainerIDToEntityName returns a prefixed entity name from a container ID
 func ContainerIDToEntityName(cid string) string {
+	if cid == "" {
+		return ""
```

Comment:
Could we return an error instead ? 
It is called here https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/events.go#L36 and then creates a tag https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/containers/docker_events.go#L65 but I could not find where the empty string would error out (https://github.com/DataDog/datadog-agent/blob/master/pkg/tagger/global.go#L24) as opposed to gathering all the tags from the empty cids.

Reply:
My reasoning would be that adding that kind of check might drift to "why does this method not return an error if the container ID I give it is not running on that host?" As it's used for every dsd packet it has to be fast.

As "" is the default string empty value, it should indeed be ignored by the tagger (it represents an unknown/invalid entity). I'll add a check `if entity == "", return err` in tagger.Tag

Reply:
@CharlyF https://github.com/DataDog/datadog-agent/pull/734

Reply:
👍